### DISCUSSION TITLE
Linux Readlink functionality in bt script

### DISCUSSION THREAD

**Discussion by DineshI-MS** [bpftrace/bpftrace#3349] [General]
I am trying to use Linux Readlink functionality inside a bt script. From the manual, I do not see any functionality for that. Is there any suggestion on using it inside a bt script?

**Comment by fbs** [comment]
Can you explain more about your usecase?

**Comment by DineshI-MS** [comment]
I am trying to get the executable path of the process using pid - `/proc/$pid/exe`

**Comment by danobi** [comment]
Hi, we currently don't have support for readlink() inside the bpf program itself. As an alternative, you could shell out and do it in userspace using system(): https://github.com/bpftrace/bpftrace/blob/master/man/adoc/bpftrace.adoc#functions-system

**Reply by DineshI-MS** [reply]
Thanks for the suggestion. I am trying to do it in the safe mode. As system() can only be used in unsafe mode, checking for other options
